src/ch/blinkenlights/android/vanilla/FileSystemAdapter.java

```diff
@@ -105,7 +128,24 @@ public int compare(File a, File b)
 			boolean aIsFolder = a.isDirectory();
 			boolean bIsFolder = b.isDirectory();
 			if (bIsFolder == aIsFolder) {
-				return a.getName().compareToIgnoreCase(b.getName());
+				int mode = aIsFolder ? SORT_NAME : getSortModeIndex();
+				int order;
+				switch (mode) {
+					case SORT_SIZE:
+						order = Long.valueOf(a.length()).compareTo(Long.valueOf(b.length()));
+						break;
+					case SORT_TIME:
+						order = Long.valueOf(a.lastModified())
+								.compareTo(Long.valueOf(b.lastModified()));
+						break;
+					case SORT_EXT:
+						order = FileUtils.getFileExtension(a.getName())
+								.compareToIgnoreCase(FileUtils.getFileExtension(b.getName()));
+						break;
+					default:
+						order = a.getName().compareToIgnoreCase(b.getName());
```

Comment:
Please add an explicit case for SORT_NAME and have 'default' throw an IllegalArgumentException()